source/QuantityUGens/QuantityUGens.cpp

```
@@ -75,8 +75,8 @@ void MovingAverage_Ctor( MovingAverage* unit )
 
 inline void MovingXCtorHelper( MovingX* unit, UnitCalcFunc calcFunc )
 {
-    if ((int) ZIN0(2) == 0) {
-        printf("MovingSum Error: maxSamps initialized to 0.\n");
+    if ((int) ZIN0(2) <= 0) {
```

that was a good catch @telephon . also FYI this is undefined behavior when the value is +/-inf, NaN, or has an integral part larger than INT_MAX. whether or not you want to care about those cases or just let it blow up is up to you.